docs/decisions/0004-external-event-bus-and-django-signal-events.rst

```diff
@@ -0,0 +1,55 @@
+4. External event bus and Django Signal events
+==============================================
+
+Status
+------
+
+Provisional
+
+Context
+-------
+
+Open edX services already use Django signals for internal events (see `OEP-50: Hooks extension framework <https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0050-hooks-extension-framework.html>`_). Additionally, we are adding the ability to trigger external events using an Event Bus to the platform (see `OEP-52: Event Bus <https://github.com/openedx/open-edx-proposals/pull/233>`_).
+
+It remained to clearly define the relationship between the new external events and the existing internal events.
+
+Decision
+--------
+
+- Event definitions will be shared between internal and external events. The event definitions will continue to be in the form of ``OpenEdxPublicSignal``, as decided in ADR ":doc:`0003-events-payload`".
+
+- At this time, event bus events will be triggered from a Django signal handler of the Django signal representing the corresponding internal event.
+
+- The Django signal must contain all event envelope and data to be sent across the event bus, such that it can be converted back into a Django signal again on the consumer side.
+
+- For consumption, the event bus implementation will convert the messages back into Django signals and emit them within the consumer application.
+
+Consequences
+------------
+
+- The ``OpenEdxPublicSignal`` serves as the event definition, and doubles as a Django signal, for both internal and external events. In other words, the ``OpenEdxPublicSignal`` provides the schema for external events.
+
+- An external event will never be sent without a corresponding internal event (at this time, based on the current design).
+
+- The external event bus handler will listen for relevant django signals (OpenEdxPublicSignals), serialize them for the event bus, and then send the messages over the wire. Serialization will use the AvroAttrsBridge, as decided in ADR ":doc:`0005-external-event-schema-format`".
```

Comment:
I feel like this line implies that all OpenEdxPublicSignals will automatically be serialized onto the event bus. If that is the decision we are making, that's fine. I just wasn't clear on it.

Reply:
Hopefully the update is more clear.